tests/e2e/default/reports/search-reports.wdio-spec.js

```diff
@@ -1,114 +1,73 @@
-  from: PHONE_NO,
-  fields: {
-    patient_id: potuPerson.patient_id
-  },
-  sms_message: {
-    message_id: 23,
-    from: PHONE_NO,
-    message: `1!RR!${potuPerson.patient_id}`,
-    form: 'RR',
-    locale: 'en'
-  },
-  reported_date: moment().subtract(10, 'minutes').valueOf()
-};
-
-const docs = [...places.values(), sittuHospital, sittuPerson, potuHospital, potuPerson, REPORT1, REPORT2];
-
-describe('Test Reports Search Functionality', async () => {
   before(async () => {
-    await utils.saveDocs(docs);
+    await utils.saveDocs([ sittuHospital, sittuPerson, potuHospital, potuPerson ]);
+    reportDocs = await utils.saveDocs(reports);
     await loginPage.cookieLogin();
-    await commonPage.goToReports();
   });
 
-  it('search by NON empty string should display results with contains match and then clears', async () => {
-    // Waiting for initial load
-    await reportsPage.getAllReportsText();
+  it('should return results matching the search term and then return all data when clearing search', async () => {
+    await commonPage.goToReports();
+    // Asserting first load reports
+    expect((await reportsPage.reportsListDetails()).length).to.equal(4);
 
-    // Searching by keyword
     await searchPage.performSearch('sittu');
-    let allLHSContacts = await reportsPage.getAllReportsText();
-    expect(allLHSContacts.sort()).to.deep.equal([sittuPerson.name]);
+    await commonPage.waitForLoaders();
+    expect((await reportsPage.reportsListDetails()).length).to.equal(2);
+    expect(await (await reportsPage.reportByUUID(reportDocs[0].id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(reportDocs[2].id)).isDisplayed()).to.be.true;
 
-    // Clearing
     await searchPage.clearSearch();
-    allLHSContacts = await reportsPage.getAllReportsText();
-    expect(allLHSContacts.sort()).to.deep.equal([potuPerson.name, sittuPerson.name]);
+    expect((await reportsPage.reportsListDetails()).length).to.equal(4);
+    expect(await (await reportsPage.reportByUUID(reportDocs[0].id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(reportDocs[1].id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(reportDocs[2].id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(reportDocs[3].id)).isDisplayed()).to.be.true;
+  });
+
+  it('should return results when searching by case_id', async () => {
+    await commonPage.goToReports();
+    await reportsPage.openReport(reportDocs[2].id);
+    // Asserting first load reports
+    expect((await reportsPage.reportsListDetails()).length).to.equal(4);
+
+    await reportsPage.clickOnCaseId();
+    await commonPage.waitForLoaders();
+    expect((await reportsPage.reportsListDetails()).length).to.equal(2);
+    expect(await (await reportsPage.reportByUUID(reportDocs[2].id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(reportDocs[3].id)).isDisplayed()).to.be.true;
```

Comment:
I think we should add a test for mobile as well, because we expect the user to be taken from the report detail page to the report list when clicking on the `case_id` field on mobile.